{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": "<polyfuzz.polyfuzz.PolyFuzz at 0x18dd09ccb80>"
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from polyfuzz import PolyFuzz\n",
    "\n",
    "from_list = [\"apple\", \"apples\", \"appl\", \"recal\", \"house\", \"similarity\"]\n",
    "to_list = [\"apple\", \"apples\", \"mouse\"]\n",
    "\n",
    "model = PolyFuzz(\"TF-IDF\")\n",
    "model.match(from_list, to_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "source": [
    "model.get_matches()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "outputs": [
    {
     "data": {
      "text/plain": "         From      To  Similarity\n0       apple   apple       1.000\n1      apples  apples       1.000\n2        appl   apple       0.784\n3       recal    None       0.000\n4       house   mouse       0.588\n5  similarity    None       0.000",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>From</th>\n      <th>To</th>\n      <th>Similarity</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>apple</td>\n      <td>apple</td>\n      <td>1.000</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>apples</td>\n      <td>apples</td>\n      <td>1.000</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>appl</td>\n      <td>apple</td>\n      <td>0.784</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>recal</td>\n      <td>None</td>\n      <td>0.000</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>house</td>\n      <td>mouse</td>\n      <td>0.588</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>similarity</td>\n      <td>None</td>\n      <td>0.000</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.get_matches()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ASUS\\AppData\\Local\\Temp\\ipykernel_13636\\1240681812.py:1: DtypeWarning: Columns (8,14,15,16,17,22,27,28,36,38,39) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  loinc_df = pd.read_csv(\"./dataset/Loinc_2.73/LoincTable/Loinc.csv\")\n"
     ]
    }
   ],
   "source": [
    "loinc_df = pd.read_csv(\"./dataset/Loinc_2.73/LoincTable/Loinc.csv\")"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "outputs": [
    {
     "data": {
      "text/plain": "      LOINC_NUM                                          COMPONENT PROPERTY  \\\n0      100002-5                        Specimen care is maintained     Find   \n1      100003-3   Team communication is maintained throughout care     Find   \n2      100004-1  Demonstrates knowledge of the expected psychos...     Find   \n3      100005-8  Demonstrates knowledge of nutritional manageme...     Find   \n4      100006-6    Demonstrates knowledge of medication management     Find   \n...         ...                                                ...      ...   \n99074   99994-6  Fluid, electrolyte, and acid-base balances are...     Find   \n99075   99995-3  Respiratory status is maintained at or improve...     Find   \n99076   99996-1  Cardiovascular status is maintained at or impr...     Find   \n99077   99997-9     Demonstrates &or reports adequate pain control     Find   \n99078   99998-7  Neurological status is maintained at or improv...     Find   \n\n      TIME_ASPCT    SYSTEM SCALE_TYP METHOD_TYP        CLASS  \\\n0             Pt  ^Patient       Ord        NaN  SURVEY.PNDS   \n1             Pt  ^Patient       Ord        NaN  SURVEY.PNDS   \n2             Pt  ^Patient       Ord        NaN  SURVEY.PNDS   \n3             Pt  ^Patient       Ord        NaN  SURVEY.PNDS   \n4             Pt  ^Patient       Ord        NaN  SURVEY.PNDS   \n...          ...       ...       ...        ...          ...   \n99074         Pt  ^Patient       Ord        NaN  SURVEY.PNDS   \n99075         Pt  ^Patient       Ord        NaN  SURVEY.PNDS   \n99076         Pt  ^Patient       Ord        NaN  SURVEY.PNDS   \n99077         Pt  ^Patient       Ord        NaN  SURVEY.PNDS   \n99078         Pt  ^Patient       Ord        NaN  SURVEY.PNDS   \n\n      VersionLastChanged CHNG_TYPE  ... COMMON_ORDER_RANK COMMON_SI_TEST_RANK  \\\n0                   2.72       ADD  ...                 0                   0   \n1                   2.72       ADD  ...                 0                   0   \n2                   2.72       ADD  ...                 0                   0   \n3                   2.72       ADD  ...                 0                   0   \n4                   2.72       ADD  ...                 0                   0   \n...                  ...       ...  ...               ...                 ...   \n99074               2.72       ADD  ...                 0                   0   \n99075               2.72       ADD  ...                 0                   0   \n99076               2.72       ADD  ...                 0                   0   \n99077               2.72       ADD  ...                 0                   0   \n99078               2.72       ADD  ...                 0                   0   \n\n       HL7_ATTACHMENT_STRUCTURE  EXTERNAL_COPYRIGHT_LINK PanelType  \\\n0                           NaN                      NaN       NaN   \n1                           NaN                      NaN       NaN   \n2                           NaN                      NaN       NaN   \n3                           NaN                      NaN       NaN   \n4                           NaN                      NaN       NaN   \n...                         ...                      ...       ...   \n99074                       NaN                      NaN       NaN   \n99075                       NaN                      NaN       NaN   \n99076                       NaN                      NaN       NaN   \n99077                       NaN                      NaN       NaN   \n99078                       NaN                      NaN       NaN   \n\n      AskAtOrderEntry AssociatedObservations VersionFirstReleased  \\\n0                 NaN                    NaN                 2.72   \n1                 NaN                    NaN                 2.72   \n2                 NaN                    NaN                 2.72   \n3                 NaN                    NaN                 2.72   \n4                 NaN                    NaN                 2.72   \n...               ...                    ...                  ...   \n99074             NaN                    NaN                 2.72   \n99075             NaN                    NaN                 2.72   \n99076             NaN                    NaN                 2.72   \n99077             NaN                    NaN                 2.72   \n99078             NaN                    NaN                 2.72   \n\n      ValidHL7AttachmentRequest DisplayName  \n0                           NaN         NaN  \n1                           NaN         NaN  \n2                           NaN         NaN  \n3                           NaN         NaN  \n4                           NaN         NaN  \n...                         ...         ...  \n99074                       NaN         NaN  \n99075                       NaN         NaN  \n99076                       NaN         NaN  \n99077                       NaN         NaN  \n99078                       NaN         NaN  \n\n[99079 rows x 41 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>LOINC_NUM</th>\n      <th>COMPONENT</th>\n      <th>PROPERTY</th>\n      <th>TIME_ASPCT</th>\n      <th>SYSTEM</th>\n      <th>SCALE_TYP</th>\n      <th>METHOD_TYP</th>\n      <th>CLASS</th>\n      <th>VersionLastChanged</th>\n      <th>CHNG_TYPE</th>\n      <th>...</th>\n      <th>COMMON_ORDER_RANK</th>\n      <th>COMMON_SI_TEST_RANK</th>\n      <th>HL7_ATTACHMENT_STRUCTURE</th>\n      <th>EXTERNAL_COPYRIGHT_LINK</th>\n      <th>PanelType</th>\n      <th>AskAtOrderEntry</th>\n      <th>AssociatedObservations</th>\n      <th>VersionFirstReleased</th>\n      <th>ValidHL7AttachmentRequest</th>\n      <th>DisplayName</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>100002-5</td>\n      <td>Specimen care is maintained</td>\n      <td>Find</td>\n      <td>Pt</td>\n      <td>^Patient</td>\n      <td>Ord</td>\n      <td>NaN</td>\n      <td>SURVEY.PNDS</td>\n      <td>2.72</td>\n      <td>ADD</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.72</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>100003-3</td>\n      <td>Team communication is maintained throughout care</td>\n      <td>Find</td>\n      <td>Pt</td>\n      <td>^Patient</td>\n      <td>Ord</td>\n      <td>NaN</td>\n      <td>SURVEY.PNDS</td>\n      <td>2.72</td>\n      <td>ADD</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.72</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>100004-1</td>\n      <td>Demonstrates knowledge of the expected psychos...</td>\n      <td>Find</td>\n      <td>Pt</td>\n      <td>^Patient</td>\n      <td>Ord</td>\n      <td>NaN</td>\n      <td>SURVEY.PNDS</td>\n      <td>2.72</td>\n      <td>ADD</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.72</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>100005-8</td>\n      <td>Demonstrates knowledge of nutritional manageme...</td>\n      <td>Find</td>\n      <td>Pt</td>\n      <td>^Patient</td>\n      <td>Ord</td>\n      <td>NaN</td>\n      <td>SURVEY.PNDS</td>\n      <td>2.72</td>\n      <td>ADD</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.72</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>100006-6</td>\n      <td>Demonstrates knowledge of medication management</td>\n      <td>Find</td>\n      <td>Pt</td>\n      <td>^Patient</td>\n      <td>Ord</td>\n      <td>NaN</td>\n      <td>SURVEY.PNDS</td>\n      <td>2.72</td>\n      <td>ADD</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.72</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>99074</th>\n      <td>99994-6</td>\n      <td>Fluid, electrolyte, and acid-base balances are...</td>\n      <td>Find</td>\n      <td>Pt</td>\n      <td>^Patient</td>\n      <td>Ord</td>\n      <td>NaN</td>\n      <td>SURVEY.PNDS</td>\n      <td>2.72</td>\n      <td>ADD</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.72</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>99075</th>\n      <td>99995-3</td>\n      <td>Respiratory status is maintained at or improve...</td>\n      <td>Find</td>\n      <td>Pt</td>\n      <td>^Patient</td>\n      <td>Ord</td>\n      <td>NaN</td>\n      <td>SURVEY.PNDS</td>\n      <td>2.72</td>\n      <td>ADD</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.72</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>99076</th>\n      <td>99996-1</td>\n      <td>Cardiovascular status is maintained at or impr...</td>\n      <td>Find</td>\n      <td>Pt</td>\n      <td>^Patient</td>\n      <td>Ord</td>\n      <td>NaN</td>\n      <td>SURVEY.PNDS</td>\n      <td>2.72</td>\n      <td>ADD</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.72</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>99077</th>\n      <td>99997-9</td>\n      <td>Demonstrates &amp;or reports adequate pain control</td>\n      <td>Find</td>\n      <td>Pt</td>\n      <td>^Patient</td>\n      <td>Ord</td>\n      <td>NaN</td>\n      <td>SURVEY.PNDS</td>\n      <td>2.72</td>\n      <td>ADD</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.72</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>99078</th>\n      <td>99998-7</td>\n      <td>Neurological status is maintained at or improv...</td>\n      <td>Find</td>\n      <td>Pt</td>\n      <td>^Patient</td>\n      <td>Ord</td>\n      <td>NaN</td>\n      <td>SURVEY.PNDS</td>\n      <td>2.72</td>\n      <td>ADD</td>\n      <td>...</td>\n      <td>0</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>2.72</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n  </tbody>\n</table>\n<p>99079 rows Ã— 41 columns</p>\n</div>"
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loinc_df"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "outputs": [
    {
     "data": {
      "text/plain": "['Specimen care is maintained',\n 'Team communication is maintained throughout care',\n 'Demonstrates knowledge of the expected psychosocial responses to the procedure',\n 'Demonstrates knowledge of nutritional management related to the procedure',\n 'Demonstrates knowledge of medication management',\n 'Demonstrates knowledge of pain management',\n 'R wave duration.lead AVR',\n 'Demonstrates knowledge of wound management',\n 'Demonstrates knowledge of the procedure and expected results',\n 'No injury related to procedure equipment, medical supplies, or instrumentation',\n 'No injury related to an electrical source',\n \"Participates in decisions affecting the patient's plan of care\",\n 'Participates in the discharge process',\n 'Psychosocial health is maintained at or improved from baseline',\n 'No injury related to positioning due to care or procedure',\n 'R wave duration.lead I',\n 'No injury related to a laser source',\n 'Perioperative nursing data set outcomes panel',\n 'Note',\n 'ALK gene targeted mutation analysis',\n 'GNA11 gene targeted mutation analysis',\n 'GNAQ gene targeted mutation analysis',\n 'IDH1 gene targeted mutation analysis',\n 'IDH2 gene targeted mutation analysis',\n 'R wave duration.lead II',\n 'SETBP1 gene targeted mutation analysis',\n 'SRSF2 gene targeted mutation analysis',\n 'MET gene targeted mutation analysis',\n 'SMAD4 gene targeted mutation analysis',\n 'FBXW7 gene targeted mutation analysis',\n 'Cancer related multigene analysis',\n 'Axis I: Temporal aspects score',\n 'Axis II: Spatial aspects score',\n 'R wave duration.lead III',\n 'Axis III: Drug taking behavior score',\n 'Axis IV: Utilization of the health care system score',\n 'Provider of automated external defibrillator use',\n 'Exam finding location',\n 'Exam finding location',\n 'Patient contact disposition',\n 'Patient evaluation disposition',\n 'Crew disposition for patient care',\n 'R wave duration.lead V1',\n 'Patient transport disposition',\n 'Aminoglycoside',\n 'Atovaquone',\n 'Azithromycin+Ethambutol',\n 'Cefcapene',\n 'Cefozopran',\n 'Cefquinome',\n 'Cefteram',\n 'Clarithromycin+Ethambutol',\n 'Eravacycline',\n 'Erythromycin+Ethambutol',\n 'Ethambutol+rifAMPin',\n 'Flomoxef',\n 'Fluoroquinolone',\n 'Gamithromycin',\n 'Optochin',\n 'Panipenem',\n 'R wave duration.lead V2',\n 'Prothionamide',\n 'Prulifloxacin',\n 'Rifapentine',\n 'Tildipirosin',\n 'Tosufloxacin',\n 'Standard Dermatology Outcome Measures panel',\n 'Primary skin concern',\n 'Itch severity',\n 'R wave duration.lead V3',\n 'Satisfied with treatment',\n 'Specular microscopy panel',\n 'Image magnification',\n 'Fixation point',\n 'Fixation point',\n 'Cell density',\n 'Cell density',\n 'Coefficient of variation',\n 'R wave duration.lead V4',\n 'Coefficient of variation',\n 'Hexagonality',\n 'Hexagonality',\n 'Endothelial cells counted',\n 'Endothelial cells counted',\n 'Endothelial cell area.min',\n 'Endothelial cell area.min',\n 'Endothelial cell area.max',\n 'R wave duration.lead V5',\n 'Endothelial cell area.max',\n 'Endothelial cell area.mean',\n 'Endothelial cell area.mean',\n 'Cell area.standard deviation',\n 'Cell area.standard deviation',\n 'Analysis method',\n 'Toxocara canis 24-35kD Ab.IgG',\n 'Taenia solium larva Ab bands panel',\n 'Taenia solium larva 6-8kD Ab',\n 'DBG Ab',\n 'Taenia solium larva 45kD Ab',\n 'Trypanosoma cruzi Ab',\n 'Trypanosoma cruzi Ab bands panel',\n 'Trypanosoma cruzi 15-16kD Ab.IgG',\n 'Trypanosoma cruzi 21-22kD Ab.IgG',\n 'Trypanosoma cruzi 27-28kD Ab.IgG',\n 'Trypanosoma cruzi 42kD Ab.IgG',\n 'Trypanosoma cruzi 45-47kD Ab.IgG',\n 'Trypanosoma cruzi 120-200kD Ab.IgG',\n 'R wave duration.lead V6',\n 'Trypanosoma cruzi 160kD Ab.IgG',\n 'Fasciola sp Ab.IgG',\n 'Fasciola sp 8-9kD Ab.IgG',\n 'Fasciola sp 27-28kD Ab.IgG',\n 'Fasciola sp 60kD Ab.IgG',\n 'Fasciola sp 42kD Ab.IgG',\n 'Filaria Ab.IgG & IgM panel',\n 'Filaria Ab.IgG',\n \"R' wave amplitude.lead AVF\",\n 'Leishmania sp Ab.IgG',\n 'Leishmania sp Ab.IgG',\n 'Leishmania sp Ab bands panel',\n 'Leishmania sp 14kD Ab.IgG',\n 'Leishmania sp 16kD Ab.IgG',\n 'Fasciola sp Ab bands panel',\n 'Helicobacter pylori Ab panel',\n 'Toxoplasma gondii 30kD Ab',\n \"R' wave amplitude.lead AVL\",\n 'Toxoplasma gondii 31kD Ab',\n 'Toxoplasma gondii 33kD Ab',\n 'Toxoplasma gondii 40kD Ab',\n 'Toxoplasma gondii 41kD Ab',\n 'Toxoplasma gondii 45kD Ab',\n 'Chlamydia trachomatis Ab panel',\n 'Chlamydia trachomatis Ab.IgA',\n 'Chlamydophila pneumoniae Ab panel',\n \"R' wave amplitude.lead AVR\",\n 'Chlamydophila pneumoniae Ab.IgA',\n 'Chlamydophila pneumoniae Ab.IgA',\n 'Chlamydophila psittaci Ab panel',\n 'Bordetella pertussis Ab.IgG panel',\n 'Campylobacter sp Ab panel',\n 'Cryptococcus sp Ag panel',\n 'Schistosoma sp Ab',\n 'Schistosoma mansoni Ab',\n \"R' wave amplitude.lead I\",\n 'Schistosoma sp 8kD Ab',\n 'Schistosoma sp 9kD Ab',\n 'Schistosoma sp 10kD Ab',\n 'Schistosoma sp 11kD Ab',\n 'Schistosoma sp 12-13kD Ab',\n 'Schistosoma sp 14-15kD Ab',\n 'Schistosoma sp 15-16kD Ab',\n 'Schistosoma sp 18-19kD Ab',\n 'Schistosoma sp 22-24kD Ab',\n 'Schistosoma sp 30-34kD Ab',\n 'Schistosoma sp 65kD Ab',\n 'Schistosoma sp 70kD Ab',\n 'Schistosoma sp 80kD Ab',\n 'Schistosoma sp 95kD Ab',\n 'Schistosoma sp 110kD Ab',\n 'Schistosoma sp 120kD Ab',\n 'Toxoplasma gondii Ab bands panel',\n 'Schistosoma sp Ab bands panel',\n \"R' wave amplitude.lead II\",\n '6-oxo-piperidine-2-carboxylate & 6(R+S)-oxo-propylpiperidine-2-carboxylate panel',\n '6(R+S)-oxo-propylpiperidine-2-carboxylate',\n '6-oxo-piperidine-2-carboxylate',\n '6-oxo-piperidine-2-carboxylate/Creatinine',\n '6(R+S)-oxo-propylpiperidine-2-carboxylate/Creatinine',\n 'Collection supervision level',\n 'Analysis method',\n \"R' wave amplitude.lead III\",\n 'SARS coronavirus 2 variant',\n 'SARS coronavirus 2 lineage',\n 'Albumin',\n 'Knee Society Score pre-op panel',\n 'Knee replacement',\n 'Charnley Functional Classification',\n 'Alignment^W standing',\n 'Medial &or lateral instability^W full extension',\n \"R' wave amplitude.lead V1\",\n 'Anterior-posterior instability^W 90 degree flexion',\n 'Range of motion',\n 'Flexion contracture range',\n 'Extensor lag range',\n 'Knee pain severity with level walking',\n 'Knee pain severity with stairs or inclines',\n 'DBG Ab',\n 'Knee feels normal',\n 'Satisfaction with level of knee pain while sitting',\n \"R' wave amplitude.lead V2\",\n 'Satisfaction with level of knee pain while lying in bed',\n 'Satisfaction with knee function while getting out of bed',\n 'Satisfaction with knee function while performing light household duties',\n 'Satisfaction with knee function while performing recreational activities',\n 'Expect surgery to relieve pain',\n 'Expect surgery will improve ability to perform activities of daily living',\n 'Expect surgery will improve ability to perform leisure, recreational, or sports activities',\n 'Walk without assistive mobility devices',\n \"R' wave amplitude.lead V3\",\n 'Assistive mobility devices used',\n 'Assistive mobility devices used due to health condition',\n 'Duration of standing before sitting due to knee discomfort',\n 'Duration of walking before stopping due to knee discomfort',\n 'How bothersome is knee when walking on an uneven surface',\n 'How bothersome is knee when turning or pivoting leg',\n 'How bothersome is knee when climbing up or down a flight of stairs',\n 'How bothersome is knee when getting up from a low seat without armrests',\n 'How bothersome is knee when getting into or out of a car',\n 'How bothersome is knee when moving laterally',\n 'How bothersome is knee when climbing a ladder or step stool',\n 'How bothersome is knee when carrying a bag for one block',\n 'How bothersome is knee when squatting',\n 'How bothersome is knee when kneeling',\n 'How bothersome is knee when running',\n 'Pain severity^during activity',\n 'Pre-op total score',\n 'Exercise activity & pain severity panel',\n \"R' wave amplitude.lead V4\",\n 'Pre-op objective knee indicators score',\n 'Pre-op symptoms score',\n 'Pre-op patient expectations score',\n 'Pre-op functional activities score',\n 'Pre-op discretionary knee activities score',\n 'Knee Society Score post-op panel',\n 'Accuracy of pre-surgery expectations of pain relief',\n 'Accuracy of pre-surgery expectations to perform activities of daily living',\n \"R' wave amplitude.lead V5\",\n 'Accuracy of pre-surgery expectations to perform leisure, recreational, or sports activities',\n 'Post-op total score',\n 'Post-op objective knee indicators score',\n 'Post-op symptoms score',\n 'Post-op patient expectations score',\n 'Post-op functional activities score',\n 'Post-op discretionary knee activities score',\n 'Prostate cancer multigene analysis',\n \"R' wave amplitude.lead V6\",\n 'Dental model prior authorization',\n 'Episode of care medical records',\n 'Synoptic report',\n 'Synoptic report',\n 'Synoptic report',\n 'Synoptic report',\n 'Synoptic report',\n 'Synoptic report',\n \"R' wave duration.lead AVF\",\n 'Synoptic report',\n 'Physical findings',\n 'Segmental wall motion panel',\n 'Bone density quantitative ultrasound study',\n 'Small bowel capsule endoscopy study',\n 'Colon capsule endoscopy study',\n 'Upper gastrointestinal capsule endoscopy study',\n 'Endoscopic ultrasound study',\n \"R' wave duration.lead AVL\",\n 'Routine prenatal assessment panel',\n 'Bronchial endoscopic ultrasound study',\n 'Impacts of Events Scale-Revised panel',\n 'Reminders brought back feelings about adverse event',\n 'Trouble staying asleep',\n 'Other things triggered persistent thoughts about adverse event',\n 'Felt irritable &or angry',\n 'Avoided getting upset when unintentionally thought about adverse event',\n 'Thought about adverse event unintentionally',\n 'Felt as if adverse event did not happen',\n 'Avoided reminders of adverse event',\n 'Experienced mental images of adverse event',\n 'Easily startled',\n 'Tried to not think of adverse event',\n 'Aware of feelings about adverse event, but did not address them',\n 'Feelings about adverse event were numb',\n 'Acted or felt the same as during adverse event',\n 'Trouble falling asleep',\n \"R' wave duration.lead AVR\",\n 'Waves of strong feelings about adverse event',\n 'Tried to remove adverse event from memory',\n 'DBG Ab',\n 'Trouble concentrating',\n 'Reminders of adverse event caused physical reactions',\n 'Had dreams about adverse event',\n 'Felt watchful and on-guard',\n 'Tried not to talk about adverse event',\n \"R' wave duration.lead I\",\n 'Total Score',\n 'Fallen in last 6Mo',\n 'Feel unsteady when standing or walking',\n 'History of fall related injury',\n 'Aware of tactile sensations',\n 'Easy to describe thoughts &or feelings',\n 'Critical of own thoughts &or feelings',\n 'Able to notice distressing thoughts &or feelings without having to react',\n \"R' wave duration.lead II\",\n 'Easily distracted',\n 'Able to notice thoughts &or feelings without being overwhelmed',\n 'Aware of dietary intake impact on self',\n 'Difficult to describe thoughts &or feelings',\n 'Aware of auditory stimuli',\n 'Able to avoid immediate reaction in difficult situations',\n 'Do things without paying attention',\n 'Recover quickly from distressing thoughts &or images',\n \"R' wave duration.lead III\",\n 'Aware of olfactory stimuli',\n 'Aware of visual stimuli',\n 'Aware of emotions affect on thoughts &or behavior',\n 'Observing score',\n 'Describing score',\n 'Acting with awareness score',\n 'Nonjudging score',\n 'Nonreactivity score',\n 'Total score',\n 'Five Facet Mindfulness Questionnaire panel',\n 'D Ab',\n 'Vaccine exemption certificate',\n 'Harris Hip Score panel',\n 'Walking - functional ability',\n 'Comfortable seated position',\n 'Able to board public transportation',\n 'Put on shoes and socks',\n 'Absence of deformity score',\n \"R' wave duration.lead V1\",\n 'Absence of hip deformity measurements',\n 'Range of motion score',\n 'Total Score',\n 'Pre-op patient satisfaction score',\n 'Flexion range of motion',\n 'Abduction range of motion',\n 'Adduction range of motion',\n 'External rotation range of motion',\n \"R' wave duration.lead V2\",\n 'Internal rotation range of motion',\n 'Repetition count',\n 'Post-op patient satisfaction score',\n 'End time',\n 'Start time',\n 'Time period start & end panel',\n 'Condition certainty of presence',\n 'Flights climbed',\n \"R' wave duration.lead V3\",\n 'IDH1 gene exon 4 targeted mutation analysis',\n 'IDH2 gene exon 4 targeted mutation analysis',\n 'Behavioral screening elder mistreatment',\n 'Need help getting to essential places',\n 'Activity support person',\n 'Reliability of support person',\n 'Need help to make sure there is enough food, medicines or any other things needed in house',\n 'Person or persons that makes sure there is enough food, medicines or any other things needed in house',\n \"R' wave duration.lead V4\",\n 'Need help with household things such as cook meals, help feed, or provide correct medicines each day',\n 'Need help with house cleaning or yard work',\n 'Need help getting out of bed, showered, or dressed',\n 'Need help to make sure bills get paid',\n 'Relationship of usual support person for daily activities',\n 'Have help with finances &or financial decisions',\n 'Relationship of financial support person',\n 'Financial support person usually asks for permission before making decisions',\n \"R' wave duration.lead V5\",\n 'Financial support person usually makes good decisions about finances',\n 'Have access to paperwork for financial decisions made',\n 'Financial support person forged signature',\n 'Coercion of financial support person for signing a document in order to get money or possessions',\n 'Experienced theft by support person',\n 'Money spent or property sold without permission',\n 'Stranger forged signature for financial gain',\n 'Stranger forced document signature for financial gain',\n 'Feared for safety due to verbal attack',\n 'Indirect antiglobulin test.complement specific reagent',\n 'Felt humiliated due to ridicule',\n 'Coerced to do something',\n 'Ignored by close friend or relative for extended period',\n 'Physically hit or threatened',\n 'Been physically hurt with some degree of injury',\n 'Coerced sexual activity',\n 'Coerced touching of private body parts of self or others',\n 'Coerced to undress or expose private body area',\n 'Photographed in any degree of nudity without consent',\n \"R' wave duration.lead V6\",\n 'Experienced attempted physical restraint',\n 'Range of motion panel',\n 'Rubella virus Ab.IgG index',\n 'Volatile Organic Compounds associated with SARS-CoV-2 infection',\n 'Influenza virus B RNA',\n 'Influenza virus A RNA',\n 'Influenza virus A & Influenza virus B & SARS coronavirus 2 RNA panel',\n 'Clotting time.extrinsic coagulation system activated.platelets inhibited',\n 'S wave amplitude.lead AVF',\n 'Clot formation.extrinsic coagulation system activated.platelets inhibited',\n 'Model for end-stage liver disease sodium score',\n 'Multisection^W vasodilator IV',\n 'Sesamum indicum 1 Ab.IgE',\n 'Pioglitazone',\n 'Rosiglitazone',\n 'Norwalk Community Health Center Screening Tool',\n 'S wave amplitude.lead AVL',\n 'Frequency of participating in multi-modal communication without assistance',\n 'Frequency of participating in spoken language expression communication without assistance',\n 'Chlamydia trachomatis & Neisseria gonorrhoeae & Trichomonas vaginalis DNA',\n 'Naloxone',\n 'Phentermine',\n 'Gliadin 33 mer peptide',\n 'Brief Resilience Scale panel',\n 'Quickly recover from bad events',\n 'S wave amplitude.lead AVR',\n 'Difficulty making it through stressful events',\n 'Easily recover from a stressful event',\n 'Difficulty recovering from bad events',\n 'Easily recover from difficulties',\n 'Lengthy recovery from setbacks',\n 'Total Score',\n 'Amino Acids Urea Cycle Panel',\n 'Views^W Tl-201 IV',\n 'S wave amplitude.lead I',\n 'Orthopoxvirus DNA',\n 'Titin Ab',\n 'Abiraterone',\n 'Eucheuma Ab.IgG',\n 'Aloe vera Ab.IgG',\n 'Pimpinella anisum Ab.IgG',\n 'Bambusa vulgaris Ab.IgG',\n 'Goat cheese Ab.IgG',\n 'Sheep cheese Ab.IgG',\n 'Allium ampeloprasum Ab.IgG',\n 'Brassica rubra Ab.IgG',\n 'Mentha balsamea Wild Ab.IgG',\n 'REMS addressed risk',\n 'Monkeypox virus DNA',\n 'Brassica rapa Ab.IgG',\n 'Valerianella locusta Ab.IgG',\n 'Rosmarinus officinalis Ab.IgG',\n 'Papaver somniferum Ab.IgG',\n 'S wave amplitude.lead II',\n 'Theobroma cacao Ab.IgG',\n 'Pistacia vera Ab.IgG',\n 'Helianthus annuus seed Ab.IgG',\n 'Phoenix dactylifera Ab.IgG',\n 'Duck meat Ab.IgG',\n 'Allergen Mushroom mix 2 (Xerocomus, Boletus) Ab.IgG',\n 'Astacoidea Ab.IgG',\n 'Ceratonia siliqua Ab.IgG',\n 'S wave amplitude.lead III',\n 'Brassica napus Ab.IgG',\n 'Capra aegagrus hircus Ab.IgG',\n 'Goose meat Ab.IgG',\n 'Ostrich meat Ab.IgG',\n 'Quail meat Ab.IgG',\n 'Rabbit meat Ab.IgG',\n 'Venison meat Ab.IgG',\n 'Guineafowl Ab.IgG',\n 'S wave amplitude.lead V1',\n 'Horse meat Ab.IgG',\n 'Kefir Ab.IgG',\n 'Butter Ab.IgG',\n 'Camembert cheese Ab.IgG',\n 'Emmental cheese Ab.IgG',\n 'Processed cheese Ab.IgG',\n 'Direct antiglobulin test.complement specific reagent',\n 'Curd cheese Ab.IgG',\n 'Beta vulgaris Ab.IgG',\n 'S wave amplitude.lead V2',\n 'Chinese cabbage Ab.IgG',\n 'Foeniculum vulgare fresh Ab.IgG',\n 'Cucurbitaceae genera Ab.IgG',\n 'Brassica oleracea var borecole Ab.IgG',\n 'Brassica oleracea var. Sabauda Ab.IgG',\n 'Vitis sp leaf Ab.IgG',\n 'Glycyrrhiza glabra Ab.IgG',\n 'Pisum sativum var. saccharatum Ab.IgG',\n 'S wave amplitude.lead V3',\n 'Vicia faba Ab.IgG',\n 'Cicer arietinus Ab.IgG',\n 'Cichorium intybus Ab.IgG',\n '(Agaricus bisporus+Pleurotus ostreatus+Lentinula edodes+Cantharellus) Ab.IgG',\n '8(9)-Cholestenol',\n 'DiHydro T-Mas',\n 'Squalene',\n 'Stigmasterol',\n 'Carnitine Free & Total & Acylcarnitine Panel',\n 'Lipopolysaccharide-responsive beige-like anchor protein deficiency panel',\n 'Cells.LRBA+/100 Cells.CD3+CD14-CD45+',\n 'Cells.CD3+CD14-CD45+.LRBA mean fluorescence intensity',\n 'Cells.LRBA+/100 Cells.CD3-CD14-CD19+CD45+',\n 'Cells.CD3-CD14-CD19+CD45+.LRBA mean fluorescence intensity',\n 'Orthopoxvirus.non-variola DNA',\n 'Glial fibrillary acidic protein',\n 'Ubiquitin carboxyl-terminal hydrolase-L1',\n 'Norsufentanil',\n 'S wave amplitude.lead V4',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave amplitude.lead V5',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave amplitude.lead V6',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'COVID-19+Intubation+Progress note',\n 'COVID-19+Progress note',\n 'Progress note',\n 'S wave duration.lead AVF',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave duration.lead AVL',\n 'Progress note',\n 'Progress note',\n 'Mechanical circulatory support+progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave duration.lead AVR',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave duration.lead I',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave duration.lead II',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave duration.lead III',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave duration.lead V1',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave duration.lead V2',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave duration.lead V3',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave duration.lead V4',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave duration.lead V5',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'S wave duration.lead V6',\n 'Progress note',\n 'Progress note',\n 'Indirect antiglobulin test.IgG specific reagent',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n \"S' wave amplitude.lead AVF\",\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Mechanical circulatory support+progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n \"S' wave amplitude.lead AVL\",\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n \"S' wave amplitude.lead AVR\",\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n \"S' wave amplitude.lead I\",\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n \"S' wave amplitude.lead II\",\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n \"S' wave amplitude.lead III\",\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n \"S' wave amplitude.lead V1\",\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n 'Progress note',\n \"S' wave amplitude.lead V2\",\n 'COVID-19+Intubation+Progress note',\n 'Pure tone bone conduction threshold audiometry panel',\n 'Pure tone air conduction threshold audiometry panel',\n 'Pyruvate kinase M2',\n 'View lateral',\n 'Progress note',\n 'Progress note',\n 'COVID-19+Progress note',\n 'Progress note',\n 'Direct antiglobulin test.IgG specific reagent',\n 'Progress note',\n 'Progress note',\n 'Cortisol free & total panel',\n 'TPMT gene c.460G>A&c.719A>G',\n 'NUDT15 gene c.50_55dup&c.415C>T',\n 'NUDT15 gene c.415C>T',\n 'NUDT15 gene c.416G>A',\n 'NUDT15 gene c.52G>A',\n 'NUDT15 gene c.50_55dup',\n \"S' wave amplitude.lead V3\",\n 'NUDT15 gene c.50_55del',\n 'RHD gene allele^fetus',\n 'Aspergillus sp DNA',\n 'Aspergillus fumigatus DNA',\n 'Aspergillus terreus DNA',\n 'BK virus DNA',\n 'Cytomegalovirus DNA',\n 'Cytomegalovirus DNA',\n \"S' wave amplitude.lead V4\",\n 'Epstein Barr virus DNA',\n 'Epstein Barr virus DNA',\n 'Herpes virus 6 DNA',\n 'Herpes virus 6 DNA',\n 'Herpes virus 7 DNA',\n 'Herpes virus 7 DNA',\n 'Herpes virus 8 DNA',\n 'Herpes virus 8 DNA',\n \"S' wave amplitude.lead V5\",\n 'JC virus DNA',\n 'JC virus DNA',\n 'JC virus DNA',\n 'Parvovirus B19 DNA',\n 'Toxoplasma gondii DNA',\n 'Toxoplasma gondii DNA',\n 'Varicella zoster virus DNA',\n 'Varicella zoster virus DNA',\n \"S' wave amplitude.lead V6\",\n 'Pneumocystis jiroveci DNA',\n 'SMN1 gene silent carrier analysis',\n 'Genetic disease analysis overall carrier interpretation',\n 'CTNS related multigene analysis',\n 'Note',\n 'Social Determinants of Health screening report',\n 'Rifabutin+Ethambutol',\n 'Trichomonas vaginalis DNA',\n \"S' wave duration.lead AVF\",\n 'Trichomonas vaginalis rRNA',\n 'Mycoplasma genitalium rRNA',\n 'Mycoplasma genitalium rRNA',\n 'Mycoplasma genitalium rRNA',\n 'Mycoplasma genitalium rRNA',\n 'Mycoplasma genitalium rRNA',\n 'Polymorphonuclear cells',\n 'Plasma cells/100 leukocytes',\n 'Hepatocellular carcinoma risk',\n 'Chlamydia trachomatis & Neisseria gonorrhoeae rRNA panel',\n 'Borrelia burgdorferi.VlsE1+pepC10 Ab',\n 'Brexpiprazole',\n 'Cariprazine',\n 'Guanfacine',\n 'Prothipendyl',\n 'Prostate Specific Ag',\n 'FCGR3A gene.p.Phe176Val',\n 'Discharge summary note',\n \"S' wave duration.lead AVL\",\n 'Discharge summary note',\n 'Discharge summary note',\n 'Discharge summary note',\n 'Discharge summary note',\n 'Discharge summary note',\n 'Discharge summary note',\n 'Discharge summary note',\n 'Discharge summary note',\n \"S' wave duration.lead AVR\",\n 'Discharge summary note',\n 'Discharge summary note',\n 'Discharge summary note',\n 'Discharge summary note',\n 'Discharge summary note',\n 'Cholesterol.in lipoprotein (little a) & Cholesterol.in LDL panel',\n 'Cholesterol.in LDL^corrected for Cholesterol.in lipoprotein (little a)',\n 'traMADol & metabolites panel',\n \"S' wave duration.lead I\",\n 'Mannose-6-phosphate isomerase & phosphomannomutase 1 panel',\n 'Acylcarnitine Analysis Post Mortem',\n 'Alkaline phosphatase.macromolecular',\n 'Alanine aminotransferase.macromolecular',\n 'Aspartate aminotransferase.macromolecular',\n 'Direct antiglobulin test.polyspecific reagent',\n 'Amylase.macromolecular',\n 'Gamma glutamyl transferase.macromolecular',\n 'Lactate dehydrogenase.macromolecular',\n \"S' wave duration.lead II\",\n 'Lipase.macromolecular',\n 'Alpha-1-Acid glycoprotein',\n 'Chloride',\n 'Glucose',\n 'Legionella serogroup panel',\n 'Legionella spp',\n 'Legionella pneumophila serogroup 1',\n \"S' wave duration.lead III\",\n 'Legionella pneumophila non serogroup 1',\n 'Galactose-alpha-1,3-galactose panel',\n 'Exacerbating factors',\n 'Alleviating factors',\n 'Platelet disorders multigene analysis',\n 'Protein fractions.oligoclonal bands',\n 'Protein fractions.oligoclonal bands',\n 'Protein fractions.oligoclonal bands panel',\n 'Estimated lung age',\n 'Inspiratory reserve',\n 'View',\n 'PALB2 gene targeted mutation analysis',\n 'HBA2 gene.c.377T>C',\n 'Mycobacterium sp comparative reaction interpretation',\n 'Mycobacterium bovis reaction wheal^3D post dose mammalian tuberculin ID',\n 'Homocysteine & Methylmalonate & 2-Methylcitrate panel',\n 'Center for Epidemiologic Studies Depression Scale panel',\n 'Bothered by things that are not usually bothersome',\n \"S' wave duration.lead V1\",\n 'Poor appetite',\n 'Unhappy even with help from my family or friends',\n 'Feeling just as good as others',\n 'Trouble focusing',\n 'Feeling depressed',\n 'Feeling everything is too much of an effort',\n 'Feeling hopeful about the future',\n 'Feeling life had been a failure',\n \"S' wave duration.lead V2\",\n 'Feeling fearful',\n 'Restless sleep',\n 'Feeling happy',\n 'Quieter than usual',\n 'Feeling lonely',\n 'People were unfriendly',\n 'Enjoying life',\n 'Crying spells',\n \"S' wave duration.lead V3\",\n 'Feeling sad',\n 'Feeling people dislike me',\n 'Lack of energy',\n 'Total Score',\n 'Stemphylium botryosum Ab.IgG4',\n 'Rhizopus nigricans Ab.IgG4',\n 'Lepidoglyphus destructor Ab.IgG',\n 'Hamster epithelium Ab.IgG',\n \"S' wave duration.lead V4\",\n 'Forsythia Ab.IgE',\n 'Testosterone',\n 'Follitropin',\n 'S100 calcium binding protein B',\n 'Thyroglobulin',\n 'Testosterone.free^1st specimen post XXX challenge',\n 'Testosterone.free^2nd specimen post XXX challenge',\n '21-Deoxycortisol^1st specimen post XXX challenge',\n 'Cefoperazone',\n \"S' wave duration.lead V5\",\n '21-Deoxycortisol^2nd specimen post XXX challenge',\n '21-Deoxycortisol^3rd specimen post XXX challenge',\n '21-Deoxycortisol^4th specimen post XXX challenge',\n '21-Deoxycortisol^5th specimen post XXX challenge',\n 'Vasopressin^1st specimen post XXX challenge',\n 'Vasopressin^2nd specimen post XXX challenge',\n 'Vasopressin^3rd specimen post XXX challenge',\n 'Vasopressin^4th specimen post XXX challenge',\n 'Vasopressin^5th specimen post XXX challenge',\n 'Vasopressin^6th specimen post XXX challenge',\n 'Vasopressin^7th specimen post XXX challenge',\n 'Vasopressin^8th specimen post XXX challenge',\n 'Vasopressin^9th specimen post XXX challenge',\n 'Vasopressin^10th specimen post XXX challenge',\n 'Vasopressin^11th specimen post XXX challenge',\n 'Vasopressin^12th specimen post XXX challenge',\n 'Glucagon^6th specimen post XXX challenge',\n 'Glucagon^7th specimen post XXX challenge',\n \"S' wave duration.lead V6\",\n 'Renin^1st specimen post XXX challenge',\n 'Renin^2nd specimen post XXX challenge',\n 'Indirect antiglobulin test.polyspecific reagent',\n 'Renin^3rd specimen post XXX challenge',\n 'POLST panel',\n 'Cardiopulmonary resuscitation orders',\n 'Initial portable medical treatment orders',\n 'Additional portable medical orders or instructions',\n 'Medically assisted nutrition orders',\n 'ST initial amplitude 6 ms.lead AVF',\n 'Portable medical order &or advance directive review',\n 'Portable medical order discussion participants',\n 'Portable medical order administrative information',\n 'Renin^4th specimen post XXX challenge',\n 'Alpha subunit.free^1st specimen post XXX challenge',\n 'Alpha subunit.free^2nd specimen post XXX challenge',\n 'Alpha subunit.free^3rd specimen post XXX challenge',\n 'Alpha subunit.free^4th specimen post XXX challenge',\n 'ST initial amplitude 6 ms.lead AVL',\n 'Alpha subunit.free^5th specimen post XXX challenge',\n 'Alpha subunit.free^6th specimen post XXX challenge',\n 'Alpha subunit.free^7th specimen post XXX challenge',\n 'Insulin-like growth factor-I^9th specimen post XXX challenge',\n 'Testosterone^9th specimen post XXX challenge',\n 'Vitamin A/Retinol binding protein',\n 'IgE',\n 'Zinc finger protein of the cerebellum 4 Ab',\n 'ST initial amplitude 6 ms.lead AVR',\n 'Nonalcoholic steatohepatitis score',\n 'Liver fibrosis score',\n 'Liver fibrosis score',\n 'Neurofascin155 Ab.IgG4',\n 'Dengue virus 1+2+3+4 non-structural protein 1 (nsP1) Ab.IgG',\n 'Gas & ammonia panel',\n 'Cell count & Differential panel',\n 'Leukocytes',\n 'Neutrophils',\n 'Other cells',\n 'Views for metastasis^W Sr-89 IV',\n 'Pericardial effusion',\n 'Lymphocytes/100 Cells',\n 'Cells.CD3/100 cells',\n 'Cells.CD3+CD4+/100 cells',\n 'Cells.CD3+CD8+/100 cells',\n 'Cells.CD4+CD154+/Cells.CD4^after stimulation with SARS-CoV-2 Spike peptide',\n 'ST initial amplitude 6 ms.lead I',\n 'Cells.CD4.Interferon gamma-expressing/Cells.CD4^after stimulation with SARS-CoV-2 Spike peptide',\n 'Cells.CD4.Tumor necrosis factor alfa-expressing/Cells.CD4^after stimulation with SARS-CoV-2 Spike peptide',\n 'Cells.CD4+CD154+/Cells.CD4^after stimulation with SARS-CoV-2 Nucleocapsid peptide',\n 'Cells.CD4.Interferon gamma-expressing/Cells.CD4^after stimulation with SARS-CoV-2 Nucleocapsid peptide',\n 'Cells.CD4.Tumor necrosis factor alfa-expressing/Cells.CD4^after stimulation with SARS-CoV-2 Nucleocapsid peptide',\n 'Giardia lamblia bg gene',\n 'Avian schistosomes DNA',\n 'Febrile antibody profile',\n 'ST initial amplitude 6 ms.lead II',\n 'Campylobacter jejuni+coli+lari+upsaliensis Ag',\n 'Cryptosporidium sp 18S-LC2 rRNA',\n 'Cryptosporidium gp60 gene',\n 'Cryptosporidium 18S rRNA',\n 'Giardia lamblia gdh gene',\n 'Bacteria producing polysaccharide from sucrose',\n 'Giardia lamblia tpi gene',\n 'Basidiobolus sp',\n 'ST initial amplitude 6 ms.lead III',\n 'Multisection limited',\n 'Multisection',\n 'Multisection transesophageal^during surgery',\n 'Endoscopic ultrasound study',\n 'Guidance for cryoablation',\n 'Guidance for cryoablation',\n 'Guidance for cryoablation',\n 'Multisection^W contrast percutaneous intraperitoneal',\n 'ST initial amplitude 6 ms.lead V1',\n 'Guidance for percutaneous biopsy',\n 'Guidance for percutaneous biopsy',\n 'Parapoxvirus DNA',\n 'Orf virus DNA',\n 'Pseudocowpox virus DNA',\n 'West African monkeypox virus DNA',\n 'Congo Basin monkeypox virus DNA',\n 'Poxvirus DNA panel',\n 'ST initial amplitude 6 ms.lead V2',\n 'Orthopoxvirus Ab.IgG',\n 'Orthopoxvirus Ab.IgM',\n 'Orthopoxvirus IgG & IgM Ab panel',\n 'Recommended screening frequency',\n 'Rickettsia conorii Ab.IgG & IgM',\n 'Vasopressin^13th specimen post XXX challenge',\n 'Staphylococcus aureus',\n 'Lipid panel',\n 'Enterobacteriaceae.extended spectrum beta lactamase resistance phenotype',\n 'Direct antiglobulin test.poly specific reagent',\n 'Enterobacteriaceae.carbapenem resistance panel',\n ...]"
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loinc_componet_list =    loinc_df['COMPONENT'].tolist()\n",
    "loinc_componet_list"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "outputs": [],
   "source": [
    "lab_test_attr = pd.read_csv(\"./dataset/Choose_Atribute/final_attribute.csv\")"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "outputs": [],
   "source": [
    "lab_test_attr_list = lab_test_attr['attr_name'].tolist()"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "outputs": [],
   "source": [
    "from sklearn.datasets import fetch_20newsgroups\n",
    "from sklearn.feature_extraction.text import CountVectorizer\n",
    "from polyfuzz import PolyFuzz"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "outputs": [],
   "source": [
    "train_words =  loinc_componet_list\n",
    "\n",
    "unseen_words  = lab_test_attr_list"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "outputs": [
    {
     "data": {
      "text/plain": "76639"
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(unseen_words)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "outputs": [
    {
     "data": {
      "text/plain": "99079"
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(train_words)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "outputs": [
    {
     "ename": "MemoryError",
     "evalue": "Unable to allocate 7.76 GiB for an array with shape (2081854878,) and data type int32",
     "output_type": "error",
     "traceback": [
      "\u001B[1;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[1;31mMemoryError\u001B[0m                               Traceback (most recent call last)",
      "Input \u001B[1;32mIn [17]\u001B[0m, in \u001B[0;36m<cell line: 3>\u001B[1;34m()\u001B[0m\n\u001B[0;32m      1\u001B[0m \u001B[38;5;66;03m# Fit\u001B[39;00m\n\u001B[0;32m      2\u001B[0m model \u001B[38;5;241m=\u001B[39m PolyFuzz(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mTF-IDF\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n\u001B[1;32m----> 3\u001B[0m \u001B[43mmodel\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mfit\u001B[49m\u001B[43m(\u001B[49m\u001B[43mtrain_words\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m      5\u001B[0m \u001B[38;5;66;03m# Transform\u001B[39;00m\n\u001B[0;32m      6\u001B[0m results \u001B[38;5;241m=\u001B[39m model\u001B[38;5;241m.\u001B[39mtransform(unseen_words)\n",
      "File \u001B[1;32me:\\workspace\\std-clinical\\lib\\site-packages\\polyfuzz\\polyfuzz.py:202\u001B[0m, in \u001B[0;36mPolyFuzz.fit\u001B[1;34m(self, from_list, to_list)\u001B[0m\n\u001B[0;32m    156\u001B[0m \u001B[38;5;28;01mdef\u001B[39;00m \u001B[38;5;21mfit\u001B[39m(\u001B[38;5;28mself\u001B[39m, \n\u001B[0;32m    157\u001B[0m         from_list: List[\u001B[38;5;28mstr\u001B[39m],\n\u001B[0;32m    158\u001B[0m         to_list: List[\u001B[38;5;28mstr\u001B[39m] \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;01mNone\u001B[39;00m):\n\u001B[0;32m    159\u001B[0m     \u001B[38;5;124;03m\"\"\" Fit one or model distance models on `from_list` if no `to_list` is given \u001B[39;00m\n\u001B[0;32m    160\u001B[0m \u001B[38;5;124;03m    or fit them on `to_list` if both `from_list` and `to_list` are given. \u001B[39;00m\n\u001B[0;32m    161\u001B[0m \u001B[38;5;124;03m    \u001B[39;00m\n\u001B[1;32m   (...)\u001B[0m\n\u001B[0;32m    200\u001B[0m \u001B[38;5;124;03m    ```\u001B[39;00m\n\u001B[0;32m    201\u001B[0m \u001B[38;5;124;03m    \"\"\"\u001B[39;00m\n\u001B[1;32m--> 202\u001B[0m     \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mmatch\u001B[49m\u001B[43m(\u001B[49m\u001B[43mfrom_list\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mto_list\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    203\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m to_list \u001B[38;5;129;01mis\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m:\n\u001B[0;32m    204\u001B[0m         \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mto_list \u001B[38;5;241m=\u001B[39m to_list\n",
      "File \u001B[1;32me:\\workspace\\std-clinical\\lib\\site-packages\\polyfuzz\\polyfuzz.py:127\u001B[0m, in \u001B[0;36mPolyFuzz.match\u001B[1;34m(self, from_list, to_list, top_n)\u001B[0m\n\u001B[0;32m    125\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mmethod \u001B[38;5;129;01min\u001B[39;00m [\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mTF-IDF\u001B[39m\u001B[38;5;124m\"\u001B[39m, \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mTFIDF\u001B[39m\u001B[38;5;124m\"\u001B[39m]:\n\u001B[0;32m    126\u001B[0m     \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mmethod \u001B[38;5;241m=\u001B[39m TFIDF(min_similarity\u001B[38;5;241m=\u001B[39m\u001B[38;5;241m0\u001B[39m, top_n\u001B[38;5;241m=\u001B[39mtop_n)\n\u001B[1;32m--> 127\u001B[0m     \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mmatches \u001B[38;5;241m=\u001B[39m {\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mTF-IDF\u001B[39m\u001B[38;5;124m\"\u001B[39m: \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mmethod\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mmatch\u001B[49m\u001B[43m(\u001B[49m\u001B[43mfrom_list\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mto_list\u001B[49m\u001B[43m)\u001B[49m}\n\u001B[0;32m    128\u001B[0m \u001B[38;5;28;01melif\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mmethod \u001B[38;5;129;01min\u001B[39;00m [\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mEditDistance\u001B[39m\u001B[38;5;124m\"\u001B[39m, \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mEdit Distance\u001B[39m\u001B[38;5;124m\"\u001B[39m]:\n\u001B[0;32m    129\u001B[0m     \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mmethod \u001B[38;5;241m=\u001B[39m RapidFuzz()\n",
      "File \u001B[1;32me:\\workspace\\std-clinical\\lib\\site-packages\\polyfuzz\\models\\_tfidf.py:91\u001B[0m, in \u001B[0;36mTFIDF.match\u001B[1;34m(self, from_list, to_list, re_train)\u001B[0m\n\u001B[0;32m     69\u001B[0m \u001B[38;5;124;03m\"\"\" Match two lists of strings to each other and return the most similar strings\u001B[39;00m\n\u001B[0;32m     70\u001B[0m \n\u001B[0;32m     71\u001B[0m \u001B[38;5;124;03mArguments:\u001B[39;00m\n\u001B[1;32m   (...)\u001B[0m\n\u001B[0;32m     87\u001B[0m \u001B[38;5;124;03m```\u001B[39;00m\n\u001B[0;32m     88\u001B[0m \u001B[38;5;124;03m\"\"\"\u001B[39;00m\n\u001B[0;32m     90\u001B[0m tf_idf_from, tf_idf_to \u001B[38;5;241m=\u001B[39m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39m_extract_tf_idf(from_list, to_list, re_train)\n\u001B[1;32m---> 91\u001B[0m matches \u001B[38;5;241m=\u001B[39m \u001B[43mcosine_similarity\u001B[49m\u001B[43m(\u001B[49m\u001B[43mtf_idf_from\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mtf_idf_to\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m     92\u001B[0m \u001B[43m                            \u001B[49m\u001B[43mfrom_list\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mto_list\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m     93\u001B[0m \u001B[43m                            \u001B[49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mmin_similarity\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m     94\u001B[0m \u001B[43m                            \u001B[49m\u001B[43mtop_n\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mtop_n\u001B[49m\u001B[43m,\u001B[49m\n\u001B[0;32m     95\u001B[0m \u001B[43m                            \u001B[49m\u001B[43mmethod\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mcosine_method\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m     97\u001B[0m \u001B[38;5;28;01mreturn\u001B[39;00m matches\n",
      "File \u001B[1;32me:\\workspace\\std-clinical\\lib\\site-packages\\polyfuzz\\models\\_utils.py:95\u001B[0m, in \u001B[0;36mcosine_similarity\u001B[1;34m(from_vector, to_vector, from_list, to_list, min_similarity, top_n, method)\u001B[0m\n\u001B[0;32m     91\u001B[0m     indices \u001B[38;5;241m=\u001B[39m np\u001B[38;5;241m.\u001B[39marray(np\u001B[38;5;241m.\u001B[39mnan_to_num(np\u001B[38;5;241m.\u001B[39marray(indices, dtype\u001B[38;5;241m=\u001B[39mnp\u001B[38;5;241m.\u001B[39mfloat), nan\u001B[38;5;241m=\u001B[39m\u001B[38;5;241m0\u001B[39m), dtype\u001B[38;5;241m=\u001B[39mnp\u001B[38;5;241m.\u001B[39mint)\n\u001B[0;32m     93\u001B[0m \u001B[38;5;66;03m# Faster than knn and slower than sparse but uses more memory\u001B[39;00m\n\u001B[0;32m     94\u001B[0m \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[1;32m---> 95\u001B[0m     similarity_matrix \u001B[38;5;241m=\u001B[39m \u001B[43mscikit_cosine_similarity\u001B[49m\u001B[43m(\u001B[49m\u001B[43mfrom_vector\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mto_vector\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m     97\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m to_list \u001B[38;5;129;01mis\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m:\n\u001B[0;32m     98\u001B[0m         np\u001B[38;5;241m.\u001B[39mfill_diagonal(similarity_matrix, \u001B[38;5;241m0\u001B[39m)\n",
      "File \u001B[1;32me:\\workspace\\std-clinical\\lib\\site-packages\\sklearn\\metrics\\pairwise.py:1385\u001B[0m, in \u001B[0;36mcosine_similarity\u001B[1;34m(X, Y, dense_output)\u001B[0m\n\u001B[0;32m   1382\u001B[0m \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[0;32m   1383\u001B[0m     Y_normalized \u001B[38;5;241m=\u001B[39m normalize(Y, copy\u001B[38;5;241m=\u001B[39m\u001B[38;5;28;01mTrue\u001B[39;00m)\n\u001B[1;32m-> 1385\u001B[0m K \u001B[38;5;241m=\u001B[39m \u001B[43msafe_sparse_dot\u001B[49m\u001B[43m(\u001B[49m\u001B[43mX_normalized\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mY_normalized\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mT\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mdense_output\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43mdense_output\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m   1387\u001B[0m \u001B[38;5;28;01mreturn\u001B[39;00m K\n",
      "File \u001B[1;32me:\\workspace\\std-clinical\\lib\\site-packages\\sklearn\\utils\\extmath.py:152\u001B[0m, in \u001B[0;36msafe_sparse_dot\u001B[1;34m(a, b, dense_output)\u001B[0m\n\u001B[0;32m    150\u001B[0m         ret \u001B[38;5;241m=\u001B[39m np\u001B[38;5;241m.\u001B[39mdot(a, b)\n\u001B[0;32m    151\u001B[0m \u001B[38;5;28;01melse\u001B[39;00m:\n\u001B[1;32m--> 152\u001B[0m     ret \u001B[38;5;241m=\u001B[39m \u001B[43ma\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;241;43m@\u001B[39;49m\u001B[43m \u001B[49m\u001B[43mb\u001B[49m\n\u001B[0;32m    154\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m (\n\u001B[0;32m    155\u001B[0m     sparse\u001B[38;5;241m.\u001B[39missparse(a)\n\u001B[0;32m    156\u001B[0m     \u001B[38;5;129;01mand\u001B[39;00m sparse\u001B[38;5;241m.\u001B[39missparse(b)\n\u001B[0;32m    157\u001B[0m     \u001B[38;5;129;01mand\u001B[39;00m dense_output\n\u001B[0;32m    158\u001B[0m     \u001B[38;5;129;01mand\u001B[39;00m \u001B[38;5;28mhasattr\u001B[39m(ret, \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mtoarray\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n\u001B[0;32m    159\u001B[0m ):\n\u001B[0;32m    160\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m ret\u001B[38;5;241m.\u001B[39mtoarray()\n",
      "File \u001B[1;32me:\\workspace\\std-clinical\\lib\\site-packages\\scipy\\sparse\\_base.py:630\u001B[0m, in \u001B[0;36mspmatrix.__matmul__\u001B[1;34m(self, other)\u001B[0m\n\u001B[0;32m    627\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m isscalarlike(other):\n\u001B[0;32m    628\u001B[0m     \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mValueError\u001B[39;00m(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mScalar operands are not allowed, \u001B[39m\u001B[38;5;124m\"\u001B[39m\n\u001B[0;32m    629\u001B[0m                      \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124muse \u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124m*\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124m instead\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n\u001B[1;32m--> 630\u001B[0m \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_mul_dispatch\u001B[49m\u001B[43m(\u001B[49m\u001B[43mother\u001B[49m\u001B[43m)\u001B[49m\n",
      "File \u001B[1;32me:\\workspace\\std-clinical\\lib\\site-packages\\scipy\\sparse\\_base.py:541\u001B[0m, in \u001B[0;36mspmatrix._mul_dispatch\u001B[1;34m(self, other)\u001B[0m\n\u001B[0;32m    539\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mshape[\u001B[38;5;241m1\u001B[39m] \u001B[38;5;241m!=\u001B[39m other\u001B[38;5;241m.\u001B[39mshape[\u001B[38;5;241m0\u001B[39m]:\n\u001B[0;32m    540\u001B[0m         \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mValueError\u001B[39;00m(\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mdimension mismatch\u001B[39m\u001B[38;5;124m'\u001B[39m)\n\u001B[1;32m--> 541\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43m_mul_sparse_matrix\u001B[49m\u001B[43m(\u001B[49m\u001B[43mother\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    543\u001B[0m \u001B[38;5;66;03m# If it's a list or whatever, treat it like a matrix\u001B[39;00m\n\u001B[0;32m    544\u001B[0m other_a \u001B[38;5;241m=\u001B[39m np\u001B[38;5;241m.\u001B[39masanyarray(other)\n",
      "File \u001B[1;32me:\\workspace\\std-clinical\\lib\\site-packages\\scipy\\sparse\\_compressed.py:529\u001B[0m, in \u001B[0;36m_cs_matrix._mul_sparse_matrix\u001B[1;34m(self, other)\u001B[0m\n\u001B[0;32m    524\u001B[0m idx_dtype \u001B[38;5;241m=\u001B[39m get_index_dtype((\u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mindptr, \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mindices,\n\u001B[0;32m    525\u001B[0m                              other\u001B[38;5;241m.\u001B[39mindptr, other\u001B[38;5;241m.\u001B[39mindices),\n\u001B[0;32m    526\u001B[0m                             maxval\u001B[38;5;241m=\u001B[39mnnz)\n\u001B[0;32m    528\u001B[0m indptr \u001B[38;5;241m=\u001B[39m np\u001B[38;5;241m.\u001B[39mempty(major_axis \u001B[38;5;241m+\u001B[39m \u001B[38;5;241m1\u001B[39m, dtype\u001B[38;5;241m=\u001B[39midx_dtype)\n\u001B[1;32m--> 529\u001B[0m indices \u001B[38;5;241m=\u001B[39m \u001B[43mnp\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mempty\u001B[49m\u001B[43m(\u001B[49m\u001B[43mnnz\u001B[49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[43mdtype\u001B[49m\u001B[38;5;241;43m=\u001B[39;49m\u001B[43midx_dtype\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    530\u001B[0m data \u001B[38;5;241m=\u001B[39m np\u001B[38;5;241m.\u001B[39mempty(nnz, dtype\u001B[38;5;241m=\u001B[39mupcast(\u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mdtype, other\u001B[38;5;241m.\u001B[39mdtype))\n\u001B[0;32m    532\u001B[0m fn \u001B[38;5;241m=\u001B[39m \u001B[38;5;28mgetattr\u001B[39m(_sparsetools, \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mformat \u001B[38;5;241m+\u001B[39m \u001B[38;5;124m'\u001B[39m\u001B[38;5;124m_matmat\u001B[39m\u001B[38;5;124m'\u001B[39m)\n",
      "\u001B[1;31mMemoryError\u001B[0m: Unable to allocate 7.76 GiB for an array with shape (2081854878,) and data type int32"
     ]
    }
   ],
   "source": [
    "# Fit\n",
    "model = PolyFuzz(\"TF-IDF\")\n",
    "model.fit(train_words)\n",
    "\n",
    "# Transform\n",
    "results = model.transform(unseen_words)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}